explorer/models.py

```diff
@@ -160,15 +164,51 @@ def __init__(self, sql, title=None,is_connection_type_pii=None):
         cursor, duration = self.execute_query()
 
         self._description = cursor.description or []
-        self._data = [list(r) for r in cursor.fetchall()]
 
-        self.duration = duration
+        if not is_connection_type_pii:
+            type_code_and_column_indices_to_be_masked_dict = self.get_type_code_and_column_indices_to_be_masked_dict()
+            self._data = self.get_data_to_be_displayed(cursor, type_code_and_column_indices_to_be_masked_dict)
```

Comment:
we have discussed this, if we need the result of first function in second, we are writing this part wrong.

Reply:
Yes, resolved.